### DISCUSSION TITLE
Is the focus on style or also on layout and control?

### DISCUSSION THREAD

**Discussion by SuperstringDev** [argyleink/open-props#381] [General]
Just came across this CSS Properties framework and it looks good. When I think of what is super cool about CSS Properties I always remember the linked demo below. 

https://googlechrome.github.io/samples/css-custom-properties/index.html

Does Open Props extend to handling layout and calibrating prop values from user interactions? What I'm looking for is allowing selective CSS Properties that are used in an app to also be calibratable by the app user.

**Comment by SuperstringDev** [comment]
Sorry about my over excitement. I see now that the demo page does interact with CSS Props in a few places. I'm selecting a CSS Framework for my team and what I am hoping to do is be able to send values from JS to CSS Props in the Root section, which can then either pick through Open Prop definitions or directly control the definitions. This is so the user is able to adjust the UI directly themselves with 'softwired' CSS Prop definitions.

**Reply by argyleink** [reply]
Nice, glad you found those examples on the docsite 🙂 they do exactly what you're lookin for, demonstrate how you could provide sliders or levers for selecting values from the provided variables to adapt a design. There's definitely nothing stopping you from listing open prop values inside a select menu or as values along an input type range. You'll also be able to scale things by adjusting the root font size if you like, and more.

(sorry it took me a bit to get back to ya here, been traveling a lot)

**Comment by SuperstringDev** [comment]
That is great news! Thanks for Open Props, CSS Properties are where it is at. I have had a issue logging on to Discord, which is being resolved so I have just one more question here.

I work with CSSGrid and so I'm interested in Open Props including tokens for that. On the main Open Props page under the Overview section it has a "Coming Soon?!" list. Are those features still coming?

**Reply by argyleink** [reply]
I've prototyped a few ideas and nothing really felt solid. Here's a [grid layout mixin style prototype](https://github.com/argyleink/open-props/blob/main/src/props.layouts.css) but i just felt it was too complicated. I do keep asking folks and looking at CSS to see if there are enough patterns in layout I could name and provide as easy props, but nothing has stuck. Have thoughts / ideas you wanna share?

**Comment by SuperstringDev** [comment]
There does seem to be some momentum picking up when it comes to using CSS Custom Properties for CSSGrid layout. The original demo I encountered was the first link below and now searching brings up more solutions:

https://googlechrome.github.io/samples/css-custom-properties/index.html
https://search.brave.com/search?q=css+variables+%2B+cssgrid&source=web
https://search.brave.com/search?q=css+custom+properties+%2B+cssgrid&source=web

This is how I want and need to define and manage my apps UI layout. It does seem like a good idea!
